site/docs/view-spec.md

```diff
@@ -0,0 +1,256 @@
+  "location" : "s3n://my_company/my/warehouse/anorwood.db/common_view",
+  "current-version-id" : 1, => current / latest version of the view. ‘1’ here since this metadata was created when the view was created.
+  "properties" : {  => shows properties of the view
+    "comment" : "View captures all the data from the table" => View comment
+  },
+  "versions" : [ { => Last few versions of the view. Controlled by the table property: “version.history.num_entries”
+    "version-id" : 1,
+    "parent-version-id" : -1,
+    "timestamp-ms" : 1573518431292,
+    "summary" : {
+      "operation" : "create", => View operation that caused this metadata to be created
+    	"engineVersion" : "presto-350",=> Version of the engine that performed the operation (create / replace)
+    },
+    "representations" : [ { => SQL metadata of the view
+      "type" : "sql",
+      "sql" : "SELECT *\nFROM\n  base_tab\n", => original view SQL
+      "schema" : { => Schema of the view expressed in Iceberg types
+        "type" : "struct",
+        "fields" : [ {
+          "id" : 0,
+          "name" : "c1",
+          "required" : false,
+          "type" : "int",
+          "doc" : "" => Column comment
+        }, {
+          "id" : 1,
+          "name" : "c2",
+          "required" : false,
+          "type" : "string",
+          "doc" : ""
+        } ]
+      },
+      "sessionCatalog" : "iceberg",=> The catalog of the user session when the view was created / replaced. Used to resolve the tables used in the view definition.
+      "sessionNamespace" : [ "anorwood" ] => Namespace at the time view was created / replaced. Used similar to ‘sessionCatalog’ above.
+    } ],
+  } ],
+  "version-log" : [ { => Log of the created versions
+    "timestamp-ms" : 1573518431292,
+    "version-id" : 1
+  } ]
+}
+```
+
+The Iceberg / view library creates a new metadata json file every time the view undergoes a DDL change. This way the history of how the view evolved can be maintained. Following metadata json file was created at the end of Step 4.
+
+```
+{
+  "format-version" : 1,
+  "location" : "s3n://my_company/my/warehouse/anorwood.db/common_view",
+  "current-version-id" : 2,
+  "properties" : {  => shows properties of the view
+    "comment" : "View captures count of the data from the table"
+  },
+  "versions" : [ {
+    "version-id" : 1,
+    "parent-version-id" : -1,
+    "timestamp-ms" : 1573518431292,
+    "summary" : {
+      "operation" : "create",
+      "engineVersion" : "presto-350",
+    },
+    "representations" : [ {
+      "type" : "sql",
+      "sql" : "SELECT *\nFROM\n  base_tab\n",
+      "schema" : {
+        "type" : "struct",
+        "fields" : [ {
+          "id" : 0,
+          "name" : "c1",
+          "required" : false,
+          "type" : "int",
+          "doc" : ""
+        }, {
+          "id" : 1,
+          "name" : "c2",
+          "required" : false,
+          "type" : "string",
+          "doc" : ""
+        } ]
+      },
```

Comment:
In future, is there a plan to parse the SQL and capture _referenced tables_ by the view as part of the View Metadata?

Reply:
Not planned. What use case do you have in mind?

Reply:
Sounds interesting. For this pull request, we are trying to keep thing minimal, but we can definitely start separate threads for discussions.

As a workaround, we can store the information as view properties.

Reply:
Agree, let's discuss the expansion separately. Great PR. 👍